### PR TITLE
fix: slash_item() errors when asked for a configuration item which doesn't exist

### LINKED ISSUE #5967: Bug: slash_item() errors when asked for a configuration item which doesn't exist
### PHP Version

8.1

### CodeIgniter4 Version

4.1.9

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Linux

### Which server did you use?

cli-server (PHP built-in webserver)

### Database

_No response_

### What happened?

`slash_item` in `Sys\Common` errors when the configuration item doesn’t exist, which is contrary to expected outcome as per the documention

### Steps to Reproduce

```
<?php

namespace App\Controllers;

class Home extends BaseController
{
    public function poc()
    {
        # returns the baseURL as expected
        echo slash_item('baseURL');

        # this generates an error - detailed below
        echo slash_item('item_does_not_exist');
    }
}
```

The error that is displayed is this:-
```
ErrorException
Undefined property: Config\App::$item_does_not_exist
SYSTEMPATH/Common.php at line 987

980      *
981      * @return string|null The configuration item or NULL if
982      *                     the item doesn't exist
983      */
984     function slash_item(string $item): ?string
985     {
986         $config     = config(App::class);
987         $configItem = $config->{$item};
988 
989         if (! isset($configItem) || empty(trim($configItem))) {
990             return $configItem;
991         }
992 
993         return rtrim($configItem, '/') . '/';
994     }

```

### Expected Output

As [per the documentation](https://codeigniter.com/user_guide/general/common_functions.html#slash_item), the return should be `null` when a configuration doesn't exist

### Anything else?

I've never done a pull request before, but can attempt to do my first one if required

However I have identified what I believe to be the cause of this and here is the updated `slash_item` code for `Common.php`

```
if (! function_exists('slash_item')) {
    // Unlike CI3, this function is placed here because
    // it's not a config, or part of a config.
    /**
     * Fetch a config file item with slash appended (if not empty)
     *
     * @param string $item Config item name
     *
     * @return string|null The configuration item or NULL if
     *                     the item doesn't exist
     */
    function slash_item(string $item): ?string
    {
        $config     = config(App::class);
        
        if (property_exists($config, $item))
        {
            $configItem = $config->{$item};
        }
        
        if (! isset($configItem) || empty(trim($configItem))) {
            return null;
        }

        return rtrim($configItem, '/') . '/';
    }
}
```

### PR DESCRIPTION
**Pull request by aletoropov**
**Description**
Check object property which doesn't exist

Fixes #5967

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
Thank you for sending this PR!

See GitHub Action checks.
1. Please fix coding style.
2. Please fix the production code to pass the existing tests.

And can you add a test case for this fix?

See
- https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#php-style
- https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/pull_request.md#unit-testing

**Comment by paulbalandan** [comment]
Please don't use merge commits when updating your PR branch with the `develop` branch.

**Comment by kenjis** [comment]
How to update your PR branch:
https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#syncing-develop

**Comment by aletoropov** [comment]
How I can created test?

**Comment by kenjis** [comment]
@aletoropov You need to learn PHPUnit and unit testing.

Unit Testing Tutorial: What is, Types, Tools & Test EXAMPLE
https://www.guru99.com/unit-testing-guide.html

PHPUnit: What, Why, How? | Andy Carter
https://andy-carter.com/blog/phpunit-what-why-how

PHP Unit Testing - PHPUnit Tutorial - Full PHP 8 Tutorial - YouTube
https://www.youtube.com/watch?v=9-X_b_fxmRM

Getting Started with Version 9 of PHPUnit – The PHP Testing Framework
https://phpunit.de/getting-started/phpunit-9.html

**Comment by kenjis** [comment]
@aletoropov See https://github.com/codeigniter4/CodeIgniter4/pull/5973/commits
Your branch still has merge commits.
Could you remove these?

See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#pushing-your-branch

**Comment by aletoropov** [comment]
> @aletoropov See https://github.com/codeigniter4/CodeIgniter4/pull/5973/commits Your branch still has three merge commits. Could you remove these?
> 
> See https://github.com/codeigniter4/CodeIgniter4/blob/develop/contributing/workflow.md#pushing-your-branch
I did it

**Comment by kenjis** [comment]
@aletoropov Thanks!

**Review by kenjis — CHANGES_REQUESTED** [review]
Please fix the product code because a test failed, and add a new test for this.

```
1) CodeIgniter\CommonFunctionsTest::testSlashItem
Failed asserting that null is identical to ''.

/home/runner/work/CodeIgniter4/CodeIgniter4/tests/system/CommonFunctionsTest.php:396
phpvfscomposer:///home/runner/work/CodeIgniter4/CodeIgniter4/vendor/phpunit/phpunit/phpunit:97
```

File: tests/system/CommonFunctionsTest.php [hunk 1]
```diff
@@ -391,7 +391,7 @@ public function testReallyWritable()
     public function testSlashItem()
     {
         $this->assertSame('/', slash_item('cookiePath')); // slash already there
-        $this->assertSame('', slash_item('cookieDomain')); // empty, so untouched
+        $this->assertNull(null, slash_item('cookieDomain')); // empty, so untouched
```
Comment (kenjis) [inline tests/system/CommonFunctionsTest.php:394]: The `$cookieDomain` is empty string. The changed assertion is incorrect.
https://github.com/codeigniter4/CodeIgniter4/blob/58aa05ee8a02994bc28e631cfe5aa3130831c259/app/Config/App.php#L261

Comment (kenjis) [reply tests/system/CommonFunctionsTest.php:394]: `public static function assertNull($actual, string $message = ''): void`

`$this->assertNull(null)` does not make sense.

**Pull request closed without merging**